viewers/nerf_viewer/bridge_server/viewer_state.py

```diff
@@ -0,0 +1,92 @@
+import time
+import base64
+import pickle
+import umsgpack
+from typing import Tuple
+import numpy as np
+import cv2 as cv
+from actions import UPDATE_STATE, UPDATE_RENDER_RESULT
+from bridge_server_subprocess import run_bridge_server_as_subprocess
+from visualizer import Viewer
+
+
+class ViewerState:
+    def __init__(self):
+        # launch the bridge server
+        websocket_port = 4567
+        zmq_port = 6000
+        ip_address = "127.0.0.1"
+
+        zmq_port = run_bridge_server_as_subprocess(
+            websocket_port=websocket_port,
+            zmq_port=zmq_port,
+            ip_address=ip_address
+        )
+        websocket_url = f"ws://localhost:{websocket_port}"
+        self.viewer_url = f"https://localhost/?websocket_url={websocket_url}"
+        self.viewer = Viewer(zmq_port=zmq_port, ip_address=ip_address)
+        self.state = None
+
+    def update_scene(self) -> None:
+        """
+        update the scene using rendered results
+        """
+        while True:
+            serialized_state = umsgpack.unpackb(self.viewer.read(UPDATE_STATE))
+            self.state = pickle.loads(serialized_state)
+            self.render_image_in_viewer()
+
+    def get_resolution(self) -> Tuple[int, int]:
+        if str(self.state.resolution) == "480":
+            return 720, 480
+        elif str(self.state.resolution) == "720":
+            return 1280, 720
+        elif str(self.state.resolution) == "1080":
+            return 1920, 1080
+        else:
+            print("invalid resolution")
```

Comment:
Please check [logger example](https://github.com/openxrlab/xrprimer/blob/main/python/xrprimer/visualization/palette/point_palette.py#L39) in class.
`self.logger.error(msg)` logs the message in error level, if a file handler has been configured, the message will never be lost. After logging, by using raise ValueError, we can prevent the program from continuing to run and causing unexpected errors, which can make it easier for developers to identify the location where the error occurred.